prestashop8.x/controllers/admin/AdminTawktoController.php

```diff
@@ -239,38 +279,98 @@ public function ajaxProcessSetVisibility()
             'show_oncustom' => json_encode([]),
 
             'enable_visitor_recognition' => false,
+            'js_api_key' => '',
         ];
 
-        $options = Tools::getValue('options');
-        if (!empty($options)) {
-            $options = explode('&', $options);
-            foreach ($options as $post) {
-                list($column, $value) = explode('=', $post);
-                switch ($column) {
-                    case 'hide_oncustom':
-                    case 'show_oncustom':
-                        // replace newlines and returns with comma, and convert to array for saving
-                        $value = urldecode($value);
-                        $value = str_ireplace(["\r\n", "\r", "\n"], ',', $value);
-                        if (!empty($value)) {
-                            $value = explode(',', $value);
-                            $jsonOpts[$column] = json_encode($value);
-                        }
+        if (empty($options)) {
+            return $jsonOpts;
+        }
+
+        $options = explode('&', $options);
+        foreach ($options as $post) {
+            [$column, $value] = explode('=', $post);
```

Comment:
Check if there's a better way to parse URL encoded forms than doing it by hand
Alternative may be switching to posting JSON if possible